### ISSUE TITLE
Can't delete repeat after modifying properties

### ISSUE THREAD

**Issue by 22justinl** [musescore/MuseScore#23534]
### Issue type

Engraving bug (incorrect score rendering)

### Description with steps to reproduce

1. Add a repeat barline (only happens with "Right (end) repeat sign")
2. Modify anything in properties tab (e.g. toggle span to next staff)
3. Try to delete repeat
Actual Behavior: Repeat sign doesn't get deleted but playback doesn't repeat at the repeat sign
Expected Behavior: Repeat gets deleted both visually and in playback

### Supporting files, videos and screenshots

https://github.com/musescore/MuseScore/assets/112573237/99873a69-cb3f-4d93-8cfd-6623cee2961f

### What is the latest version of MuseScore Studio where this issue is present?

master

### Regression

No

### Operating system

macOS 13

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by bkunda** [comment]
@zacjansheski pls check regression status

**Comment by zacjansheski** [comment]
Not a regression, same behavior in MS3 as well

Also, repeat remains after save and reopen, only way to get rid of it is to actually delete the measures

**Comment by mathesoncalum** [comment]
Appears to be fixed in current master (@zacjansheski)

**Comment by ghost** [comment]
Yes, seems this issue got duplicated as https://github.com/musescore/MuseScore/issues/24534 and was fixed in 4.6 (by https://github.com/musescore/MuseScore/pull/28815).
